File: Modelos Java/ArrayJava/src/main/java/com/mycompany/arrayjava/Controlador/Controlador.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.arrayjava.Controlador;

/**
 *
 * @author 57320
 */
import com.mycompany.arrayjava.Modelo.Modelo;
import com.mycompany.arrayjava.Vista.Vista;
import java.util.Scanner;

public class Controlador {
    private Modelo modelo = new Modelo();
    private Vista vista = new Vista();

    public void ejecutar() {
        Scanner sc = new Scanner(System.in);
        System.out.print("Ingrese el número de políticos (mínimo 10): ");
        int n = sc.nextInt();
        sc.nextLine(); // Consumir newline

        if (n < 10) {
            System.out.println("El número mínimo de políticos es 10. Se establecerá n = 10 por defecto.");
            n = 10;
        }

        String metodo = vista.pedirMetodoOrdenamiento();

        modelo.generarPoliticos(n);
        vista.mostrarPoliticos(modelo.politicos, "\nLista de políticos generada:");

        modelo.ordenarPoliticos(metodo);
        vista.mostrarPoliticos(modelo.politicos, "\nLista de políticos ordenada por dinero robado:");
    }
}

File: Modelos Python/ArrayPython/controlador/controlador.py
from modelo.modelo import Modelo
from vista.vista import Vista

class Controlador:
    def __init__(self):
        self.modelo = Modelo()
        self.vista = Vista()

    def ejecutar(self):
        try:
            n = int(input("Ingrese el numero de politicos (minimo 10): "))
        except ValueError:
            n = 10

        if n < 10:
            print("El numero minimo de politicos es 10. Se establecera n = 10 por defecto.")
            n = 10

        metodo = self.vista.pedir_metodo_ordenamiento()

        self.modelo.generar_politicos(n)
        self.vista.mostrar_politicos(self.modelo.politicos, "\nLista de politicos generada:")

        self.modelo.ordenar_politicos(metodo)
        self.vista.mostrar_politicos(self.modelo.politicos, "\nLista de politicos ordenada por dinero robado:")
